aws-lambda-extension-fluent-bit/main.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os/exec"
+	"path/filepath"
+
+	"os"
+	"os/signal"
+	"path"
+	"syscall"
+
+	extension "aws-lambda-extensions/aws-lambda-extensions-fluent-bit/extensions"
+	"aws-lambda-extensions/aws-lambda-extensions-fluent-bit/logsapi"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	defaultListenerPort = "1234" // DefaultListenerPort sets the http port to 1234 if not provided by function env variable "HOST"
+	maxItems            = 10000
+	maxBytes            = 262144
+	timeOut             = 1000
+	fluentBitPath       = "/opt/extensions/fluent-bit"
+)
+
+// Subscribes Lambda Extension to Logs API
+func subscribeExtensionToLogsAPI(logsApiClient *logsapi.Client, agentID string, listenerPort string) (*logsapi.SubscribeResponse, error) {
+	eventTypes := []logsapi.EventType{logsapi.Platform, logsapi.Function}
+	bufferingCfg := logsapi.BufferingCfg{
+		MaxItems:  maxItems,
+		MaxBytes:  maxBytes,
+		TimeoutMS: timeOut,
+	}
+
+	destination := logsapi.Destination{
+		Protocol:   logsapi.HttpProto,
+		URI:        logsapi.URI(fmt.Sprintf("http://sandbox:%s", listenerPort)),
+		HttpMethod: logsapi.HttpPost,
+		Encoding:   logsapi.JSON,
+	}
+
+	resp, err := logsApiClient.Subscribe(eventTypes, bufferingCfg, destination, agentID)
+
+	return resp, err
+}
+
+func main() {
+	extensionName := path.Base(os.Args[0])
+	printPrefix := fmt.Sprintf("[%s]", extensionName)
+	logger := log.WithFields(log.Fields{"agent": extensionName})
+
+	// Determines listener port, use default 1234 if not provided as Lambda environment variable
+	listenerPort := os.Getenv("PORT")
+	if os.Getenv("PORT") == "" {
+		listenerPort = defaultListenerPort
+	}
```

Comment:
Would it be better to use another variable name rather than the generic PORT. All extensions and the function share ENVVARS so perhaps making it more specific to the extension with something such as FLB_PORT would be preferable